mono/utils/mono-dl-windows.c

```diff
@@ -68,7 +72,11 @@ mono_dl_open_file (const char *file, int flags)
 		if (!hModule)
 			SetLastError (last_error);
 	} else {
+#if G_HAVE_API_SUPPORT(HAVE_CLASSIC_WINAPI_SUPPORT)
 		hModule = GetModuleHandle (NULL);
+#else
+		g_assert(0 && "Not supported");
```

Comment:
`g_error("Not supported")`

Reply:
@luhenry: I've fixed this assert at de81fcf04812f385f7b445a965a0b845df0df381.